config/initializers/elasticsearch.rb

```diff
@@ -1,7 +1,16 @@
 module Elasticsearch; end
 
-es_config = (YAML.load_file("#{Rails.root}/config/elasticsearch.yml") || {})[Rails.env]
+config = Rails.application.config.elasticsearch
 
-Tire::Configuration.url(es_config['url']) if es_config && es_config['url'].present?
+Elasticsearch::INDEX_NAME = config && config['index_name'].present? ? config['index_name'].freeze : "#{Rails.env}:jobs".freeze
 
-Elasticsearch::INDEX_NAME = es_config && es_config['index_name'].present? ? es_config['index_name'].freeze : "#{Rails.env}:jobs".freeze
+Rails.application.config.elasticsearch_client = Elasticsearch::Client.new(
```

Comment:
Why store instance of `Elasticsearch::Client` in rails config?
Also you can do access the configurations with `Rails.configuration.elasticsearch`